### PR TITLE
Add AUTH_TYPE env

### PR DESCRIPTION
**Pull request by jonmattgray**
- [x] add AUTH_TYPE env + readme + helm
- [x] make jwt verification, OpenAPI spec, security handlers and other AUTH envs conditional on AUTH_TYPE
- [x] move auth related code to its own `util/auth.js`

### CODE REVIEW

File: app/util/auth.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        resolve(true)
+      } else {
+        logger.warn(`Error verifying jwks`)
+        reject({ message: 'An error occurred during jwks verification' })
+      }
+    })
+  })
+}
+
+const openApiSecurity = () => {
```
Comment (jonmattgray) [inline app/util/auth.js:50]: open to a better name for this

File: app/server.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ const apiDoc = require('./api-v3/api-doc')
 const apiService = require('./api-v3/services/apiService')
```
Comment (mattdean-digicatapult) [inline app/server.js:13]: In this file we instantiate:

```js
securityHandlers: {
  bearerAuth: (req) => {
    return verifyJwks(req.headers['authorization'])
  },
}
```

This probably shouldn't even be set as an option if `AUTH_TYPE` is `NONE`

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
A few small changes. Thoughts on tests?

File: app/server.js (hunk 2 above)
Comment (jonmattgray) [reply app/server.js:13]: done

File: app/util/auth.js (hunk 1 above)
Comment (jonmattgray) [reply app/util/auth.js]: changed, much better name

File: app/env.js [hunk 3]
```diff
@@ -32,6 +32,10 @@ const vars = envalid.cleanEnv(process.env, {
   SUBSTRATE_STATUS_TIMEOUT_MS: envalid.num({ default: 2 * 1000 }),
   IPFS_STATUS_POLL_PERIOD_MS: envalid.num({ default: 10 * 1000 }),
   IPFS_STATUS_TIMEOUT_MS: envalid.num({ default: 2 * 1000 }),
+  AUTH_TYPE: envalid.str({
+    default: 'NONE',
+    choices: ['NONE', 'JWT'],
+  }),
```
[the remark this answers is not in the document]
Comment (jonmattgray) [reply app/env.js]: done

**Comment by jonmattgray** [comment]
> A few small changes. Thoughts on tests?

Test changes to cover both `AUTH_TYPE` will be in the next PR

**Review by mattdean-digicatapult — APPROVED** [review]
I'm surprised this didn't break tests. I know that you're looking at that in the next PR so happy to merge this one

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]
Add a condition to the `configmap` JWKS auth stuff on if `auth.type` = JWKS or whatever the other value for the envar is.

Make that section of envars in the `deployment` dependent on the same condition.

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**